### PR TITLE
Fix ME Input Hatch and Output Bus

### LINKED ISSUE #825: ME Stocking Input Hatch Eats Fluids
### GregTech CEu Version

1.20.1-1.1.1.a

### Recipe Viewer Installed

JEI

### Environment

Singleplayer

### Cross-Mod Interaction

No

### Expected Behavior

I would be able to include an ME Stocking Input Hatch with a fluid inside, such as oxygen for a blast furnace, and it would hold the fluid until the recipe starts.

### Actual Behavior

The ME Stocking Input Hatch consumes the fluid even if no recipe is going and if there is the ingredients for a recipe in the other buses/hatches, it consumes the fluid and doesnt start a recipe.  It continues to consume the fluid until there is none left in the ME system.

### Steps to Reproduce

Setup a standard EBF and replace the input hatch with an ME Stocking Input Hatch set to oxygen and connected to an ME system with Oxygen.  I couldnt always reliable get it to happen, sometimes it would hold the fluid, but more often than not it would just consume it and do nothing.

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by StarL0st**
## What
Fixes the ME Input Hatch not pulling fluids from ae2 and not properly showing them to the multiblock, and also potentially f ix an issue with the Output Bus, particularly based on #825 and #651 respectively

## Implementation Details
Fixing the Output Bus simply implied adding a simple if check, and fixing the ME Input Hatch implied a lot of things, I changed the position of the Change Amount Box for it to be usable and attempted to fix an issue with writing on the textbox, but that didnt work, for the fluid logic it involved fixing a series of if checks that would never be true and overriding the way `NotifiableFluidTank` handles its `FluidStorages`, the particular field in the class is now private and all uses for it were changed to use the new lombok provided `getStorages()` getter

## Outcome
Fixed Me Input Hatch and potentially fixed an issue with ME Output Bus

### CODE REVIEW

File: src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEInputHatchPartMachine.java [hunk 1]
```diff
@@ -60,14 +64,20 @@ public Widget createUIWidget() {
...
+    @Override
+    public void onLoad() {
+        super.onLoad();
+        tankSubs = this.aeFluidTanks.addChangedListener(this::updateTankSubscription);
+    }
+
     @Override
     protected void autoIO() {
-        if (getLevel().isClientSide) return;
+         if (getLevel().isClientSide) return;
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEInputHatchPartMachine.java]: Extra space

File: src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEInputHatchPartMachine.java [hunk 2]
```diff
@@ -117,6 +129,17 @@ public ExportOnlyAEFluidList(MetaMachine machine, int slots, long capacity, IO i
                 this.tanks[i] = new ExportOnlyAEFluid(null, null);
                 this.tanks[i].setOnContentsChanged(this::onContentsChanged);
             }
+            this.fluidStorages = null;
+        }
+
+        @Override
+        public FluidStorage[] getStorages() {
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEInputHatchPartMachine.java:136]: Could you make the stream creation in `ExportOnlyAEFluidList#handleRecipeInner` use `getStorages()`?

File: src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEInputHatchPartMachine.java [hunk 3]
```diff
@@ -213,11 +236,10 @@ public FluidStorage copy() {
         }
     }
 
-    public static class ExportOnlyAEFluid extends ExportOnlyAESlot implements IFluidStorage, IFluidTransfer {
+        public static class ExportOnlyAEFluid extends ExportOnlyAESlot implements IFluidStorage, IFluidTransfer {
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEInputHatchPartMachine.java]: Why the indent?

**Review by screret — CHANGES_REQUESTED** [review]

File: src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEInputHatchPartMachine.java (hunk 1 above)
Comment (StarL0st) [reply src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEInputHatchPartMachine.java]: must have been a misinput

File: src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEInputHatchPartMachine.java (hunk 3 above)
Comment (StarL0st) [reply src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEInputHatchPartMachine.java]: same here, misinput

File: src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEInputHatchPartMachine.java (hunk 2 above)
Comment (StarL0st) [reply src/main/java/com/gregtechceu/gtceu/integration/ae2/machine/MEInputHatchPartMachine.java:136]: ok

**Review by screret — APPROVED** [review]

**Pull request merged**